test/e2e/jobs/jobs.test.ts

```diff
@@ -0,0 +1,149 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { GenericContainer, StartedTestContainer, Wait } from "testcontainers";
+// import { LogWaitStrategy } from "testcontainers/build/wait-strategies/log-wait-strategy";
+import { CommunicationProtocolEnum, DaprClient, DaprServer } from "../../../src";
+// import { AbstractWaitStrategy } from "testcontainers/build/wait-strategies/wait-strategy";
+
+jest.setTimeout(10000);
+
+describe("Jobs End to End", () => {
+
+    let daprScheduler: StartedTestContainer | null = null;
+    let daprd: StartedTestContainer | null = null;
+    let server: DaprServer | null = null;
+    let client: DaprClient | null = null;
+
+    beforeAll(async () => {
+
+        daprScheduler = await new GenericContainer("ghcr.io/dapr/dapr")
+          .withExposedPorts(
+            {
+              host: 8083,
+              container: 8083,
+            }
+          )
+          .withCommand([
+            "./scheduler",
+            "--listen-address", "0.0.0.0",
+            "--port", "8083",
+            "--healthz-port", "8084",
+            "--enable-metrics", "false",
+            // note: This is here because `--enable-metrics=false` doesn't seem to be working.
+            "--metrics-port", "8085",
+          ])
+          // note: This is here because of daemon visibility issues.
+          .withNetworkMode("host")
+          .withTmpFs({
+            "/data": "rw",
+          })
+          // note: Because dapr containers don't have `sh` or `bash` inside, this is kind of the best health check.
+          .withWaitStrategy(Wait.forLogMessage("api is ready").withStartupTimeout(10000))
+          .start();
+
+        daprd = await new GenericContainer("ghcr.io/dapr/dapr")
+          .withExposedPorts(
+            {
+                host: 8081,
+                container: 8081
+            },
+            {
+                host: 8082,
+                container: 8082,
+            }
+          )
+          .withCommand([
+            "./daprd",
+            "--app-id", "dapr-js-sdk-testing",
+            "--app-port", "8070",
+            "--dapr-grpc-port", "8081",
+            "--dapr-http-port", "8082",
+            "--scheduler-host-address", "0.0.0.0:8083",
+            "--placement-host-address", "",
+            "--enable-metrics", "false",
+          ])
+          // note: This is here because well, the daemon likes it.
+          .withNetworkMode("host")
+          // note: Because dapr containers don't have `sh` or `bash` inside, this is kind of the best health check.
+          .withWaitStrategy(Wait.forLogMessage("HTTP server is running on port").withStartupTimeout(10000))
+          .start()
+        ;
+    });
+
+    beforeEach(async () => {
+
+      server = new DaprServer({
+        serverHost: "127.0.0.1",
```

Comment:
Is this right? Or should this be the network alias or `host.testcontainers.internal`?

https://node.testcontainers.org/features/networking/#expose-host-ports-to-container

Reply:
You're running docker-in-docker in CI, so I doubt that will resolve to the machine running the tests.

My advice:
1. This should run in a custom network. I think this is essential.
2. This should not use fixed host ports (because it's good practice)

By the way, I'm familiar with the following tests but there might be better examples in other modules:

https://github.com/testcontainers/testcontainers-node/blob/main/packages/modules/localstack/src/localstack-container.test.ts

https://github.com/testcontainers/testcontainers-node/blob/main/packages/modules/k3s/src/k3s-container.test.ts